aes/src/ni.rs

```diff
@@ -33,23 +28,62 @@ use core::arch::x86 as arch;
 #[cfg(target_arch = "x86_64")]
 use core::arch::x86_64 as arch;
 
-use crate::{Block, Block8};
 use cipher::{
-    consts::{U16, U24, U32, U8},
-    inout::InOut,
-    AlgorithmName, BlockBackend, BlockCipher, BlockCipherDecrypt, BlockCipherEncrypt, BlockClosure,
-    BlockSizeUser, Key, KeyInit, KeySizeUser, ParBlocksSizeUser,
+    consts::{self, U16, U24, U32},
+    AlgorithmName, BlockCipher, BlockCipherDecrypt, BlockCipherEncrypt, BlockClosure,
+    BlockSizeUser, Key, KeyInit, KeySizeUser,
 };
 use core::fmt;
 
+impl_backends!(
+    enc_name = Aes128BackEnc,
+    dec_name = Aes128BackDec,
+    key_size = consts::U16,
+    keys_ty = expand::Aes128RoundKeys,
+    par_size = consts::U15,
```

Comment:
Since x86 has only 16 XMM registers (AVX-512 is out of scope for now), processing 15 blocks in parallel on x86 means that each round key will be loaded on each iteration. It maximizes ILP, but introduces additional loads from L1 cache.

For AES-128, 192, and 256 we can process only 5, 3, and 1 block in parallel respectively without reloading some keys.  On my laptop the sweet spot seems to be 11 blocks (according to the crate benchmarks), but it's likely highly dependent on CPU model. We will need additional benchmarks including the CTR mode to find optimal numbers. For now I decided to use 15 blocks for assembler simplicity.